sound/soc/codecs/rt5682-sdw.c

```diff
@@ -717,7 +717,7 @@ static int rt5682_sdw_remove(struct sdw_slave *slave)
 }
```

Comment:
Given SDW is relatively new addition to kernel, I'd add a bit more detail on the rationale and what "Use SDW_SLAVE_ENTRY" means. I.e. I can see from the code you change to SDW_SLAVE_ENTRY, but what does this mean -- I'd prefer to write this out in commit message.

In the message body, I don't see the connection between "should be used for all rt5682 version" and "no plan to support sdw 1.2". Is the intention that rt5682 does not support sdw 1.2 extensions and codec driver should reflect this?

Reply:
> Given SDW is relatively new addition to kernel, I'd add a bit more detail on the rationale and what "Use SDW_SLAVE_ENTRY" means. I.e. I can see from the code you change to SDW_SLAVE_ENTRY, but what does this mean -- I'd prefer to write this out in commit message.

@kv2019i Sorry for confusing you. The issue is that the rt5682 codec is sdw version 1.2 on TGL chromebook. However, the sdw_device_id table of rt5682-sdw codec driver list sdw version 1.1 only
``SDW_SLAVE_ENTRY_EXT(0x025d, 0x5682, ``**0x1**``, 0, 0),``
Therefore rt5682 sdw driver will not probe on TGL chromebook. If we use SDW_SLAVE_ENTRY, then sdw version will be ignore. 

> In the message body, I don't see the connection between "should be used for all rt5682 version" and "no plan to support sdw 1.2". Is the intention that rt5682 does not support sdw 1.2 extensions and codec driver should reflect this?

In general, we will have different driver to support sdw 1.2 (SDCA). That's why we need to use SDW_SLAVE_ENTRY_EXT to determine the sdw version. So, if rt5682 won't support sdw 1.2, the existing rt5682 sdw codec driver can support all rt5682 version (no other rt5682 sdw codec driver needed). And we can simply use SDW_SLAVE_ENTRY.

But a benefit of using SDW_SLAVE_ENTRY_EXT is that we don't need to change the code again if rt5682 support sdw 1.2 in the future.

Reply:
@bardliao you've got the versions mixed
0x1 is SoundWire 1.0
0x2 is SoundWire 1.2
0x3 is SoundWire 1.2

Reply:
> @bardliao you've got the versions mixed
> 0x1 is SoundWire 1.0
> 0x2 is SoundWire 1.1
> 0x3 is SoundWire 1.2

Ah, yes. I don't know why I typed this in the first paragraph. Things in my mind was sdw 1.0 and 1.1.

Reply:
Thanks @bardliao , makes sense now.